pkgs/development/tools/vcstool/default.nix

```diff
@@ -0,0 +1,21 @@
+{ stdenv, fetchurl, python35Packages }:
+
+  python35Packages.buildPythonApplication rec {
+    name = "vcstool";
+    version = "0.1.31";
+    #no tests
+    doCheck = false;
```

Comment:
@FRidh would it not be helpful to have support for `makeWrapperArgs = []` also in `nixpkgs/pkgs/development/interpreters/python/wrapper.nix`? This package for instance would need tools like `git` and `svn` in its PATH. And the only way to inject this I am seeing is to have to have the executable wrapped twice.

Reply:
Then it could look like this:

```diff
diff --git a/pkgs/development/tools/vcstool/default.nix b/pkgs/development/tools/vcstool/default.nix
index 65ce3334b8..f03db562ab 100644
--- a/pkgs/development/tools/vcstool/default.nix
+++ b/pkgs/development/tools/vcstool/default.nix
@@ -1,21 +1,35 @@
-{ stdenv, fetchurl, python35Packages }:
-
-  python35Packages.buildPythonApplication rec {
-    name = "vcstool";
-    version = "0.1.31";
-    #no tests
-    doCheck = false;
-
-    propagatedBuildInputs = with python35Packages; [ pyyaml ];
-    src = fetchurl {
-        url = "mirror://pypi/v/vcstool/${name}-${version}.tar.gz";
-        sha256 = "0n2zkvy2km9ky9lljf1mq5nqyqi5qqzfy2a6sgkjg2grvsk7abxc";
-    };
-
-    meta = with stdenv.lib; {
-        description = "Provides a command line tool to invoke vcs commands on multiple repositories";
-        homepage = "https://github.com/dirk-thomas/vcstool";
-        license = licenses.asl20;
-        maintainer = with maintainers; [ sivteck ];
-    };
+{ stdenv, fetchFromGitHub, python3Packages, makeWrapper, git }:
+
+with python3Packages;
+
+buildPythonApplication rec {
+  name = "vcstool-${version}";
+  version = "0.1.31";
+
+  src = fetchFromGitHub {
+    owner = "dirk-thomas";
+    repo = "vcstool";
+    rev = version;
+    sha256 = "0xs77frwx1kvxks57nfw90vimhiaw61dl4kyhwbmcwnjs81xzb66";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  propagatedBuildInputs = [ pyyaml git ];
+
+  makeWrapperArgs = ["--prefix" "PATH" ":" "${stdenv.lib.makeBinPath [ git ]}"];
+
+  checkInputs = [ pytest ];
+  checkPhase = ''
+    # requires pydocstyle, which is not packaged yet
+    rm test/test_flake8.py
+    py.test test
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Provides a command line tool to invoke vcs commands on multiple repositories";
+    homepage = https://github.com/dirk-thomas/vcstool;
+    license = licenses.asl20;
+    maintainer = with maintainers; [ sivteck ];
+  };
 }
```

Reply:
How is `wrapper.nix` relevant here? `buildPythonApplication` is practically the same as `buildPythonPackage`.

Reply:
yes

Reply:
What code is used to wrap python executables then?

Reply:
Could also just patch the references to the executables. e.g. in the case of git
https://github.com/dirk-thomas/vcstool/blob/0.1.31/vcstool/clients/git.py#L382

Reply:
Exactly the same as `buildPythonPackage`:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/wrap.sh